### PR TITLE
Cleaning old code

### PR DESCRIPTION
**Pull request by danielaRiesgo**
## Summary ##
Finishing issue: #82

Updating to swift 4 new features:

### Array
Leaving `group<K>(withCriteria:)` method, but now uses new `Dictionary(grouping:by:)`.

### Collection
Removing `first(where:)` that already comes with Swift 4.

### Dictionary
Removing `append` since Swift 4 introduces `merge(uniquingKeysWith:)`. Same with `appending` and `merging`.
Removing `value(forKey key:, or defaultValue:)` since now we have `[x, default: y]`.

### CODE REVIEW

File: Core/Extensions/ReactiveCocoa/Signal.swift [hunk 1]
```diff
@@ -9,17 +9,44 @@
 import ReactiveSwift
 import Result
 
+// Can't extend ResultProtocol to EventProtocol.
+// Now Result-valued signals can de materialez and dematerialized.
```
Comment (pablog) [inline Core/Extensions/ReactiveCocoa/Signal.swift]: `be materialized`?

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:x: Code Coverage (Slather): 84.17 - Minimum: 85.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/cleaning-old-code) for more details.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 86.0
:white_check_mark: Code Quality (Swiftlint): 94.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/cleaning-old-code) for more details.

**Comment by woloxci** [comment]
**CodeStats metrics details:**

:white_check_mark: Code Coverage (Slather): 86.0
:white_check_mark: Code Quality (Swiftlint): 94.0


Click [here](http://codestats.wolox.com.ar/organizations/wolox/projects/wolmo-core-ios/branches/cleaning-old-code) for more details.

**Pull request merged**